### PR TITLE
Changelog and tweaks to publish

### PR DESCRIPTION
**Pull request by lread**
- Add change log
- Add change log checking and updating on publish
- Hide tasks supporting publish
- Add println logging to publish tasks
- Add scm info to our jar

### CODE REVIEW

File: bb.edn [hunk 1]
```diff
@@ -51,16 +51,52 @@
...
+                                (System/exit 1))
+                              (println "PASS: Change log Unreleased section found with some text.")))}
 
-  tag (shell (str "git tag v" bs/version))
+  -change-log-update {:doc "Updates change log for current version"
+                      :depends [change-log-check]
+                      :task (do
+                              (println "Updating change log")
+                              (bs/refresh-version)
+                              (let [log (slurp "CHANGELOG.md")
```
Comment (borkdude) [inline bb.edn:68]: @lread Perhaps we can move some of the longer code fragments into build_shared.clj, what do you think?

Comment (lread) [reply bb.edn:68]: Thanks for the review. Sure, I think that makes sense.  I shall make it so.

Side bar: I noticed that could not use regex reader conditional `#"myregexhere"` bb in bb `:task` code. Hence the usage of `re-pattern`.

Comment (borkdude) [reply bb.edn:68]: Yep, that's expected, EDN doesn't support it.

Comment (lread) [reply bb.edn:68]: Right, of course!

**Pull request merged**